lib/cron-job/cron-job.ts

```diff
@@ -65,5 +66,13 @@ export class CronJob extends Construct {
     if (!props.schedule) {
       throw new Error("Schedule must be specified");
     }
+
+    // Validate that we get a full 5 or 6 character cron expression
+    // parseExpression only checks for max length not min length
+    if (props.schedule.split(" ").length < 5) {
+      throw new Error("Invalid cron expression");
+    }
+
+    parseExpression(props.schedule);
```

Comment:
The cron-parser library throws errors for the following conditions:

- 'Invalid cron expression' - when the expression string is longer than 6 options
- 'Constraint error'  - when the expression at a point i.e. seconds does not match the constraints available i.e. not 0-59

```
Constraint error, got value 100 expected range 0-59
```

I'm not sure if we want to throw up this error or try and parse it and return a more helpful message i.e.

```
Constraint error, got value 100 for seconds expected value in range 0-59 seconds
```

Do we also want to include anything referencing which cron job or syntax was wrong in the error? I'm thinking if there are multiple cron jobs specified for a project to know which one had the problem.

Reply:
I think including the `id` in the error message would be useful. But if we're going to catch their error and work out which unit is incorrect, then we're basically writing our own cron parser...?

Reply:
Will leave the messages as they are :) simple to start with